File: apps/api/schemas/repo_evaluator.py
"""Schemas for the standalone Repo Evaluator feature."""

from apps.api.schemas.base import BaseSchema


class TechStackInfo(BaseSchema):
    """Detected technology stack of a repository."""

    languages: list[str]
    frameworks: list[str]
    package_managers: list[str]
    databases: list[str]
    notable_deps: list[str]


class FileStats(BaseSchema):
    """LOC stats for a single file."""

    path: str
    loc: int
    language: str


class StructureInfo(BaseSchema):
    """Repository structure analysis results."""

    total_files: int
    total_loc: int
    files_over_300_loc: list[FileStats]
    hooks_over_150_loc: list[FileStats]
    directory_pattern: str  # "atomic", "feature-based", "flat", "mixed"
    has_barrel_files: bool


class PatternFinding(BaseSchema):
    """A single code pattern finding with severity assessment."""

    category: str  # "api_calls", "auth", "state_mgmt", "db_access", "css"
    pattern: str  # "raw_fetch", "repository_pattern", etc.
    occurrences: int
    example_files: list[str]
    severity: str  # "aligned", "needs_change", "critical"
    recommendation: str


class EvaluationDocument(BaseSchema):
    """A rendered markdown document section of the evaluation."""

    title: str
    slug: str
    content: str  # Markdown content
    order: int


class EvaluationResult(BaseSchema):
    """Complete evaluation output for a repository."""

    repo_path: str
    evaluated_at: str
    tech_stack: TechStackInfo
    structure: StructureInfo
    patterns: list[PatternFinding]
    has_lovable_project: bool
    documents: list[EvaluationDocument]
    summary_score: int  # 0-100 alignment score


class BrowseEntry(BaseSchema):
    """A single filesystem entry returned by the browse endpoint."""

    name: str
    path: str
    is_dir: bool


class BrowseResponse(BaseSchema):
    """Response for the directory browse endpoint."""

    current_path: str
    parent_path: str | None
    entries: list[BrowseEntry]


class EvaluateRequest(BaseSchema):
    """Request body for the evaluate endpoint."""

    repo_path: str

File: apps/web/src/lib/types/repo-evaluator.ts
export interface TechStackInfo {
  languages: string[];
  frameworks: string[];
  package_managers: string[];
  databases: string[];
  notable_deps: string[];
}

export interface FileStats {
  path: string;
  loc: number;
  language: string;
}

export interface StructureInfo {
  total_files: number;
  total_loc: number;
  files_over_300_loc: FileStats[];
  hooks_over_150_loc: FileStats[];
  directory_pattern: string;
  has_barrel_files: boolean;
}

export interface PatternFinding {
  category: string;
  pattern: string;
  occurrences: number;
  example_files: string[];
  severity: "aligned" | "needs_change" | "critical";
  recommendation: string;
}

export interface EvaluationDocument {
  title: string;
  slug: string;
  content: string;
  order: number;
}

export interface EvaluationResult {
  repo_path: string;
  evaluated_at: string;
  tech_stack: TechStackInfo;
  structure: StructureInfo;
  patterns: PatternFinding[];
  has_lovable_project: boolean;
  documents: EvaluationDocument[];
  summary_score: number;
}

export interface BrowseEntry {
  name: string;
  path: string;
  is_dir: boolean;
}

export interface BrowseResponse {
  current_path: string;
  parent_path: string | null;
  entries: BrowseEntry[];
}
